Cookbook/Technical-Documents/CRP-Bot.md

```diff
@@ -0,0 +1,101 @@
+* Even if that side effect is unintentional, we need to know that it ended up shipped in the UK release. (Even if admittedly in the ideal case the ticket should not have affected the UK release, but bugs and unintended side-effects happen)
+
+Note however that even if a US ticket will be flagged with e.g. "iOS Babylon 4.5.0" (if corresponding PR was merged before the UK 4.5.0 release), it will eventually also be flagged with e.g. the "iOS Ascension 1.1.0" version in JIRA too once it's actually shipped as part of the US release.
+
+This means that a ticket will ultimately end up with multiple versions in its "Fix Versions" field (typically for different flavors and platforms), allowing us to know all the released versions in which the code related to that ticket ended up, even if it was dormant/inactive/disabled for some flavors.
+
+Again, this might seem surprising at first, but this is expected and on purpose.
+
+### Impact on "Fix versions" vs feature flags
+
+Another impact of that process is that **even features that are gated by a local feature switch** (hardcoded as being turned off in code) and end up still being disabled when we release the app **will still be flagged with the released version in their "Fix Version"**.
+
+This is because – similar to what was described above with the example of unwanted side effects of a US feature on the UK app – the implementation of a feature that is theoretically supposed to be disabled by a feature flag can accidentally introduce bugs or have some of the code not protected behind the feature flag like it should have been. So since the code is present in the final app anyway – even if it's supposed to be disabled – the corresponding JIRA ticket will still have the version set in its "Fix Version".
+
+This isn't really an issue in the end since usually we'll end up having a separate ticket to enable the feature switch and the feature in the future. So that new JIRA ticket about turning the feature on will have its own "Fix Version" telling us when it was enabled.
+
+Still, this is a behavior to be aware of to understand the real meaning we put behind that "Fix Version" field (i.e. "the code implementing this feature ended up in the codebase shipped on the store for this version, even if supposedly disabled").
+
+### "Fix versions" and planning versions ahead
+
+Some teams might want to use JIRA versions to plan ahead, at the beginning of a sprint, which tickets we hope to be part of which version.
+
+But since there's no guarantee that those tickets will make it in time in the planned release when the corresponding release branch is cut, it would lead to inconsistencies to use the Fix Version and JIRA releases named `"<platform> <app> <version>"` to mark those planned tickets before the release is cut; because if those planned tickets end up not making the cut, you'd still have manually marked them with a Fix Version that is supposed to reflect what was _actually_ shipped as part of the release.
```

Comment:
```suggestion
But since there's no guarantee that those tickets will make it in time in the planned release when the corresponding release branch is cut, using the Fix Version and JIRA releases named `"<platform> <app> <version>"` to mark those planned tickets before the release is cut it would likely lead to inconsistencies. Additionally, these missing tickets would need to be manually updated with a `Fix Version` in order to reflect what was _actually_ shipped as part of the release.
```

Not sure I am misinterpreting the original meaning of the sentence though.

Reply:
I need to rewrite that paragraph after discussing with Mark Bates. New official suggestion for this is just to use the `iOS Babylon RC x.y.z` naming approach in Fix Version field (and not Affects Version field), I'll clarify this a bit better.

Reply:
@NSMyself thanks for the suggestions. Just rewrote the entire last paragraph about planned releases, would appreciate a new read  😉

Reply:
On it